[BUG] Circular evaluationDependsOn between :app and a prebuilt library that calls evaluationDependsOn(':app') (@notifee/react-native)

## 📋 Description

When RNRepo treats a library as a supported prebuilt, it eagerly calls `project.evaluationDependsOn(":<packageName>")` at **configuration time**. If that library's own `build.gradle` also calls `project.evaluationDependsOn(':app')`, the two constraints form a mutual evaluation cycle and Gradle aborts configuration.

This reproduces today with **`@notifee/react-native`**, whose `android/build.gradle` contains (line 46):

```groovy
if (project.hasProperty('reactNativeProjects')) {
  reactNativeProjects.each { dependent -> project.evaluationDependsOn(dependent) }
} else {
  project.evaluationDependsOn(':app')   // <-- always runs; `reactNativeProjects` is not set
}
```

The build fails with:

```
FAILURE: Build failed with an exception.
* Where:
Build file '.../node_modules/@notifee/react-native/android/build.gradle' line: 46
* What went wrong:
A problem occurred evaluating project ':notifee_react-native'.
> Circular referencing during evaluation for project ':app'.
```

The failure is **not** caused by any change in the consuming app. It appeared the moment the `@notifee/react-native@9.1.8` prebuilt AAR for RN 0.83.9 was published to `packages.rnrepo.org` (`Last-Modified: Tue, 7 Jul 2026 09:59:41 GMT`; `maven-metadata.xml` `<lastUpdated>20260707100010</lastUpdated>`). Before the prebuilt existed, notifee built from source, no cross-project `evaluationDependsOn` was introduced by RNRepo, and builds were green. Publishing the AAR flipped notifee into the supported-prebuilt set and exposed the conflict — so this can regress any consumer with no change on their side, purely from a registry publish.

Root of the cycle:
- RNRepo (`PrebuildsPlugin`): from `:app`'s configuration, eagerly `evaluationDependsOn(":notifee_react-native")` → forces `:notifee_react-native` to evaluate.
- notifee (`build.gradle:46`): during its own evaluation, `evaluationDependsOn(':app')` → forces `:app` to evaluate.
- `:app` ⇄ `:notifee_react-native` → circular reference.

### Suggested fixes (upstream)
- Avoid the eager, configuration-time `evaluationDependsOn(":<pkg>")` for supported prebuilts, or defer the task-linking so it does not require forcing the library subproject's evaluation from `:app`; and/or
- Detect a library that already declares `evaluationDependsOn(':app')` (or otherwise participates in a cycle) and skip the eager evaluation for it (fall back to source or link lazily), rather than failing the whole configuration.

### Workaround (for other users hitting this)
Add the package to the platform `denyList` in `rnrepo.config.json` to force a source build:
```json
{ "denyList": { "android": ["@notifee/react-native"] } }
```

## 🔄 Steps to Reproduce

1. Use a React Native (0.83.x) Android app with the RNRepo prebuilds Gradle plugin enabled.
2. Depend on `@notifee/react-native@9.1.8` (which ships `evaluationDependsOn(':app')` in its `android/build.gradle`) and do **not** set a `reactNativeProjects` project property.
3. Ensure the notifee prebuilt AAR is available in the registry for your RN version (it now is: `org.rnrepo.public:notifee_react-native:9.1.8:rn0.83.9@aar`), so RNRepo classifies it as a supported prebuilt.
4. Run `./gradlew :app:bundleDevGooglePlayRelease` (any `:app` assemble/bundle task).
5. Configuration fails with `Circular referencing during evaluation for project ':app'` pointing at `@notifee/react-native/android/build.gradle:46`.

Adding `@notifee/react-native` to `denyList.android` makes the failure disappear (notifee builds from source).

## 🛠️ Environment Information

**@rnrepo/build-tools version:** 0.1.4

**Environment details:**
- react-native: 0.83.9
- @notifee/react-native: 9.1.8
- Gradle: 9.0.0
- JVM: 17
- Platform: Android (fails on CI — Bitrise, ubuntu-noble-24.04 — and reproducible locally)
- RNRepo registry coordinate observed: `org.rnrepo.public:notifee_react-native:9.1.8:rn0.83.9@aar`

--- a/libraries.json
+++ b/libraries.json
@@ -969,18 +969,6 @@
       }
     ]
   },
-  "@notifee/react-native": {
-    "android": [
-      {
-        "versionMatcher": ["9.1.8", "9.0.2"]
-      }
-    ],
-    "ios": [
-      {
-        "versionMatcher": ["9.1.8", "9.0.2"]
-      }
-    ]
-  },
   "react-native-keep-awake": {
     "postInstallScriptPath": "postInstallScripts/react-native-keep-awake/react-native-keep-awake.ts",
     "android": [

